zerver/tests/test_auth_backends.py

```diff
@@ -1663,6 +1727,95 @@ def register_extra_endpoints(self, requests_mock: responses.RequestsMock,
+                                       subdomain=subdomain, is_signup=True)
+        self.stage_two_of_registration(result, realm, subdomain, email, name, name,
+                                       self.BACKEND_CLASS.full_name_validated,
+                                       user_avatar_url=mock_user_avatar_url, use_social_avatar=True,
+                                       expected_social_avatar_content_path=mocked_github_avatar_image_path)
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_register_user_and_dont_use_github_avatar(self) -> None:
+        """This tests if there is a user_avatar_url but the user chooses not to use it."""
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        mocked_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+        realm = get_realm("zulip")
+
+        mocked_github_avatar_image_path = 'zerver/tests/images/img.png'
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mocked_user_avatar_url)
+
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup=True)
+        self.stage_two_of_registration(result, realm, subdomain, email, name, name,
+                                       self.BACKEND_CLASS.full_name_validated,
+                                       user_avatar_url=mocked_user_avatar_url,
+                                       use_social_avatar=False,
+                                       expected_social_avatar_content_path=mocked_github_avatar_image_path)
+
+    @override_settings(TERMS_OF_SERVICE=None)
+    def test_github_avatar_upload_fails(self) -> None:
+        email = 'newuser@zulip.com'
+        name = 'Full Name'
+        subdomain = 'zulip'
+        mocked_user_avatar_url = 'https://github.com/user/mock_avatar_url/image.jpeg'
+        realm = get_realm("zulip")
+
+        mocked_github_avatar_image_path = 'zerver/tests/images/img.png'
+        account_data_dict = self.get_account_data_dict(email=email, name=name, user_avatar_url=mocked_user_avatar_url)
+        result = self.social_auth_test(account_data_dict,
+                                       expect_choose_email_screen=True,
+                                       subdomain=subdomain, is_signup=True)
+        self.stage_two_of_registration(result, realm, subdomain, email, name, name,
+                                       self.BACKEND_CLASS.full_name_validated,
+                                       user_avatar_url=mocked_user_avatar_url,
+                                       use_social_avatar=True,
+                                       expected_social_avatar_content_path=mocked_github_avatar_image_path,
+                                       avatar_upload_error=True)
+
+    @override_settings(TERMS_OF_SERVICE=None)
```

Comment:
Oh, just realized I all of these `@override_settings(TERMS_OF_SERVICE=None)` decorator can probably be removed from your tests, I think they're not necessary